pkg/agent/controller/networkpolicy/reconciler_test.go

```diff
@@ -439,6 +440,124 @@ func TestReconcilerReconcile(t *testing.T) {
+			Pods: appliedToGroup1,
+		},
+		{
+			rule: &rule{ID: "ingress-rule-diff-named-port", Direction: v1beta1.DirectionIn, Services: []v1beta1.Service{serviceHTTP}},
+			Pods: appliedToGroupWithDiffContainerPort,
+		},
+		{
+			rule:          &rule{ID: "egress-rule", Direction: v1beta1.DirectionOut},
+			FromAddresses: nil,
+			ToAddresses:   addressGroup1,
+			Pods:          appliedToGroup1,
+		},
+	}
+	expectedOFRules := []*types.PolicyRule{
+		{
+			Direction: v1beta1.DirectionIn,
+			From:      ipsToOFAddresses(sets.NewString("1.1.1.1")),
+			To:        ofPortsToOFAddresses(sets.NewInt32(1)),
+			Service:   []v1beta1.Service{serviceTCP80, serviceTCP},
+		},
+		{
+			Direction: v1beta1.DirectionIn,
+			From:      []types.Address{},
+			To:        ofPortsToOFAddresses(sets.NewInt32(1)),
+			Service:   nil,
+		},
+		{
+			Direction: v1beta1.DirectionIn,
+			From:      []types.Address{},
+			To:        ofPortsToOFAddresses(sets.NewInt32(1)),
+			Service:   []v1beta1.Service{serviceTCP80},
+		},
+		{
+			Direction: v1beta1.DirectionIn,
+			From:      []types.Address{},
+			To:        ofPortsToOFAddresses(sets.NewInt32(3)),
+			Service:   []v1beta1.Service{serviceTCP443},
+		},
+		{
+			Direction: v1beta1.DirectionOut,
+			From:      ipsToOFAddresses(sets.NewString("2.2.2.2")),
+			To:        ipsToOFAddresses(sets.NewString("1.1.1.1")),
+			Service:   nil,
+		},
+	}
+	tests := []struct {
+		name              string
+		args              []*CompletedRule
+		expectedOFRules   []*types.PolicyRule
+		numInstalledRules int
+		wantErr           bool
+	}{
+		{
+			"batch-install",
+			completedRules,
+			expectedOFRules,
+			0,
+			false,
+		},
+		{
+			"batch-install-partial",
+			completedRules,
+			expectedOFRules,
+			1,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+			mockOFClient := openflowtest.NewMockClient(controller)
+			r := newReconciler(mockOFClient, ifaceStore)
+			if tt.numInstalledRules > 0 {
+				// BatchInstall should skip rules already installed
+				r.lastRealizeds.Store(tt.args[0].ID, newLastRealized(tt.args[0]))
+			}
+			// TODO: mock idAllocator and priorityAssigner
+			mockOFClient.EXPECT().BatchInstallPolicyRuleFlows(gomock.Any()).
+				Do(func(rules []types.OFPolicyRule) {
+					if tt.numInstalledRules == 0 && len(rules) != len(tt.expectedOFRules) {
+						t.Fatalf("Expect to install %v flows while %v flows were installed",
+							len(tt.expectedOFRules), len(rules))
+					} else if tt.numInstalledRules > 0 && len(rules) != len(tt.expectedOFRules)-tt.numInstalledRules {
+						t.Fatalf("Expect to install %v flows while %v flows were installed",
+							len(tt.expectedOFRules)-tt.numInstalledRules, len(rules))
+					}
+				})
+			if err := r.BatchReconcile(tt.args); (err != nil) != tt.wantErr {
+				t.Fatalf("BatchReconcile() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
```

Comment:
could you use the testify `require` / `assert` statements (whichever one is appropriate for each check)  in tests when possible? For consistency and it tends to be less verbose / more readable.

Reply:
Done